public/app/ui/components/search/results/SearchResultsComponent.js

```diff
@@ -0,0 +1,59 @@
+(function () {
+    'use strict';
+
+    define(
```

Comment:
This is almost exclusively just moved out of `SearchPage`, except for the parameter checking which is only required now because of the decoupling.